client/cli/src/commands/inspect_node_key.rs

```diff
@@ -0,0 +1,75 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implementation of the `inspect-node-key` subcommand
+
+use crate::{Error, NetworkSchemeFlag};
+use std::fs;
+use libp2p::identity::{PublicKey, ed25519};
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+/// The `inspect-node-key` command
+#[derive(Debug, StructOpt)]
+#[structopt(
+	name = "inspect-node-key",
+	about = "Print the peer ID corresponding to the node key in the given file."
+)]
+pub struct InspectNodeKeyCmd {
+	/// Name of file to read the secret key from.
+	#[structopt(long)]
+	file: PathBuf,
+
+	#[allow(missing_docs)]
+	#[structopt(flatten)]
+	pub network_scheme: NetworkSchemeFlag,
+}
+
+impl InspectNodeKeyCmd {
+	/// runs the command
+	pub fn run(&self) -> Result<(), Error> {
+		let mut file_content = hex::decode(fs::read(&self.file)?)
+			.map_err(|_| "failed to decode secret as hex")?;
+		let secret = ed25519::SecretKey::from_bytes(&mut file_content)
```

Comment:
Is this command only usable for ed25519? would it make sense to generalize this so that i can also get the public key from an `sr25519` file for example? I saw the `CryptoScheme` earlier and i think it can be used here to receive an additional argument for the crypto scheme.

Reply:
I think for now, we only support ed25519, see 

https://github.com/paritytech/substrate/blob/00791403e55a9d4fc6d6a1ab8aa2fe8bb383546f/client/cli/src/arg_enums.rs#L80-L86